### DISCUSSION TITLE
Tailwind in Nuxt rc 13 does not seem to work

### DISCUSSION THREAD

**Discussion by murugappanrm** [nuxt/nuxt#16432] [Questions] [answered]
Hi,

I have tried all the tricks in the forum but somehow when i install the tailwindcss it does not seem to work. After installating Tailwindcss and autoprefixer, i changed the following:

In nuxt.conifg.ts
-----------------
```
// https://v3.nuxtjs.org/api/configuration/nuxt.config
import { defineNuxtConfig } from "nuxt/config";
export default defineNuxtConfig({
  postcss: require("./postcss.config.js"),
  css: ["assets/css/tailwind.css", "assets/css/mystyles.css"],
});
```

In postcss.config.js
-----------------
```
export const plugins = {
  tailwindcss: {},
  autoprefixer: {},
  
};
```
I am getting these errors when i run yarn dev:
```

 ERROR  [postcss] Unexpected token ':'                                                     06:47:29
 ERROR  [postcss] Unexpected token ':'                                                     06:47:29  
 ERROR  Cannot read properties of undefined (reading 'length')                             06:47:29  
 ERROR  Cannot read properties of undefined (reading 'length')                             06:47:29  
 ERROR  Cannot read properties of undefined (reading 'length')                             06:47:29  
 ERROR  Cannot read properties of undefined (reading 'length')       

```
Please help as I am a newbie on how to install tailwindcss in Nuxt rc.13

**Comment by himyjan** [comment, marked as answer]
@nuxtjs/tailwindcss https://github.com/nuxt/framework/discussions/8715#discussioncomment-4091044 (Tailwind Viewer included and support `tailwind.config.ts` config file)

# or

tailwindcss with postcss https://github.com/nuxt/framework/discussions/8715#discussioncomment-4092886

---

"@nuxtjs/tailwindcss" , and "tailwindcss"
```
npm i -D @nuxtjs/tailwindcss tailwindcss
```

[nuxt.config.ts](https://github.com/himyjan/Nuxt.js-3-RC/blob/main/nuxt.config.ts)
```
// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  modules: [
    '@nuxtjs/tailwindcss',
  ],
  typescript: {
    strict: true,
    typeCheck: {
      eslint: {
        files: './**/*.{ts,js,vue}',
      },
    },
  },
});
```

[tailwind.config.ts](https://github.com/himyjan/Nuxt.js-3-RC/blob/main/tailwind.config.ts)
```
import { Config } from 'tailwindcss';

export default <Partial<Config>>{
  content: [
    './app.{js,ts,vue}',
    './App.{js,ts,vue}',
    './error.{js,ts,vue}',
    './Error.{js,ts,vue}',
    './components/**/*.{js,vue,ts}',
    './composables/**/*.{js,ts}',
    './layouts/**/*.vue',
    './pages/**/*.vue',
    './plugins/**/*.{js,ts}',
    './nuxt.config.{js,ts}',
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```
my nuxt 3 + TailwindCSS repo - https://github.com/nuxt/framework/discussions/8715#discussioncomment-4091044
https://github.com/himyjan/Nuxt.js-3-RC

my nuxt 3 + TailwindCSS repo - pure tailwindcss (answer below https://github.com/nuxt/framework/discussions/8715#discussioncomment-4092886)
https://github.com/himyjan/Nuxt.js-3-RC/tree/tailwindcss

**Comment by murugappanrm** [comment]
Hi @himyjan 

Its ok. I resolved it without using "@nuxtjs/tailwindcss". Installing with as per tailwind starter worked fine. I was paranoid on syntax highlight error on the "defineNuxtConfig()" function. 

By the way, I tried using the  "@nuxtjs/tailwindcss" as you have suggested and it worked great too but could not use it because i wanted to use the pure tailwindcss. Thank you for all the help.

**Reply by mdr0414** [reply]
@murugappanrm 
Curious, as I am also trying to get tailwindcss to work with nuxt rc13 ... 
you said - "Installing with as per tailwind starter worked fine" - is this per the current TailwindCSS docs?
tia

**Reply by TheAlexLichter** [reply]
Easiest would be starting as in https://tailwindcss.nuxtjs.org/examples/basic

**Reply by mdr0414** [reply]
true ... but was just looking at pure tailwind - thx

**Reply by murugappanrm** [reply]
Hi @mdr0414 

Follow these steps to install pure tailwindcss:

(1) Install Nuxt 3 as per Nuxt documentation.
(2) Install Tailwindcss as follows (careful do not install postcss8 as per tailwind documentation, does not seem to work with Nuxt 3):
```
npm install -D tailwindcss postcss autoprefixer
npx tailwind init
```
(3) Change nuxt.config.ts as follows (pls ignore the vscode warning on the "defineNuxtConfig" and you do not need to import either):
```

export default defineNuxtConfig({
  postcss: {
    plugins: {
      tailwindcss: {},
      autoprefixer: {},
    },
  },
  css: ["@/assets/css/tailwind.css"],
});
```
(4) Put this code in tailwind.css file under a new root folder "assets/css":
```

@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700&display=swap");
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .btn {
    @apply inline-flex justify-center items-center whitespace-nowrap bg-red-500 hover:bg-red-400 text-white transition-colors rounded;
  }
}

```

(5) Add these codes to tailwind.config.js:
```

module.exports = {
  content: [
    "./components/**/*.{js,vue,ts}",
    "./layouts/**/*.vue",
    "./pages/**/*.vue",
    "./plugins/**/*.{js,ts}",
    "./nuxt.config.{js,ts}",
    "./**/*.vue",
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: ["Poppins", "sans-serif"],
      },
    },
  },
  plugins: [],
};

```
(6) Try this code in App.vue:

```
<template>
  <div>
    <button class="btn px-5 py-2 m-2">Hello</button>
    <NuxtWelcome />
  </div>
</template>

```

Run "yarn dev" and you should be good to go.

**Reply by mdr0414** [reply]
Thank you!  I'll give it a whirl ...
